### DISCUSSION TITLE
rocketmq SDK的使用

### DISCUSSION THREAD

**Discussion by scwlkq** [apache/rocketmq#7918] [Q&A] [answered]
![image](https://github.com/apache/rocketmq/assets/89067331/223f93e6-4b76-4382-9a71-afde985246ce)
这里的第一个字符串我看usage里是addr ，请问这个addr传namesrvAdd 还是 BrokerUrl 还是都可以哇
![image](https://github.com/apache/rocketmq/assets/89067331/539d5940-1a3a-4527-bcff-2dd5861d947a)

**Comment by pxd98** [comment, marked as answer]
这个应该是看你上层的请求，如果你的请求是往nameserver发的，应该填null，如果往broker发的，就填broker的url。
比如你这个请求是sendmessage，就应该是填brokerurl，如果你的请求是查询topic的路由信息，就应该填null。
如果仅仅是使用这个sdk的话，应该直接用类似于producer.setNamesrvAddr(“namesrvAddr”)就可以了，sdk里会自动帮你判断的；除非你是想直接用这个接口。

**Reply by scwlkq** [reply]
![image](https://github.com/apache/rocketmq/assets/89067331/6c7939dd-68c1-43ea-a8b4-37f51feac7f2)
我在使用RemotingClient的时候遇到了一些问题， 可以帮忙看下么 非常感谢

用DefaultAdminExt是可以成功的 所以mq应该没啥问题

**Reply by pxd98** [reply]
这里应该是要配broker的地址，不是nameserver的地址，还有就是remotingClient要先start
